### PR TITLE
Added proxy support for S3UploadManager

### PR DESCRIPTION
**Pull request by vthakre**
This PR is for adding the proxy settings to enable accessing AWS S3 through proxy server 

@HenryCaiHaiying

### CODE REVIEW

File: src/main/java/com/pinterest/secor/common/SecorConfig.java [hunk 1]
```diff
@@ -237,6 +237,18 @@ public String getAwsSecretKey() {
     public String getAwsEndpoint() {
         return getString("aws.endpoint");
     }
+    
+    public boolean getAwsProxyEnabled(){
+    	return getBoolean("aws.proxy.isEnabled", false);
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/common/SecorConfig.java]: We usually want the config default specified in properties file, there is no need to provide a default in java code.

**Comment by HenryCaiHaiying** [comment]
Looks fine, only a small comment on property default.

When you made the follow-up fix, can you squash two commits into one so it's cleaner in revision history.

File: src/main/java/com/pinterest/secor/common/SecorConfig.java (hunk 1 above)
Comment (vthakre) [inline src/main/java/com/pinterest/secor/common/SecorConfig.java]: for the boolean property, the method getBoolean expects 2 parameter, properyName and the defaultValue.

Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/common/SecorConfig.java]: Can you add a getBoolean method which doesn't require default value?  Or
directly calling mProperties.getBoolean()

Comment (vthakre) [inline src/main/java/com/pinterest/secor/common/SecorConfig.java]: Added a method getBoolean with just propertyName. Also Squashed the commits

**Comment by HenryCaiHaiying** [comment]
@vthakre Looks good, thanks for the patience on the iterations.

**Pull request merged**